### ISSUE TITLE
[BUG] PolynomialTrendForecaster - degree parameter not working properly

### ISSUE THREAD

**Issue by SueP12** [alan-turing-institute/sktime#8686]
**Describe the bug**

When fitting `PolynomialTrendForecaster` with different polynomial degrees, the output trend sometimes does not match the specified degree.
In the reproducible example below, degrees 2, 3, and 4 produce identical fitted curves, despite sklearn’s equivalent pipeline producing visibly different results.

**To Reproduce**

```python
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from sklearn.pipeline import make_pipeline
from sklearn.preprocessing import PolynomialFeatures
from sklearn.linear_model import LinearRegression
from sktime.forecasting.trend import PolynomialTrendForecaster
from sktime.forecasting.base import ForecastingHorizon

def create_data():
    """
    Create synthetic time series data with cubic trend: up fast, then down, then up again.
    """
    dates = pd.date_range('2019-01-01', '2024-12-31', freq='D')
    n = len(dates)
    t = np.linspace(-1, 1, n)
    trend = 1000 + 500 * t + 200 * t**2 - 300 * t**3
    noise = np.random.normal(0, 20, n)
    y = trend + noise
    df = pd.DataFrame({
        'DATE': dates,
        'VALUE': y
    })
    
    return df

def plot_polynomial_trend_comparison(
    df: pd.DataFrame,
    target_col: str = "VALUE",
    date_col: str = "DATE", 
    max_degree: int = 3,
):
   
    # Prepare data
    df = df.copy()
    df = df.set_index(date_col)
    df = df.sort_index()
    df["time_int"] = np.arange(len(df))
    
    # Convert to pandas Series for sktime
    y_series = df[target_col]

    # Create subplots: rows for degrees, columns for sklearn vs sktime
    n_plots = max_degree + 1
    fig, axs = plt.subplots(n_plots, 2, figsize=(20, 5 * n_plots), sharex=True)
    
    for degree in range(n_plots):
        ax_sklearn = axs[degree, 0]
        ax_sklearn.scatter(df.index, df[target_col], color="lightgray", s=5, alpha=0.6, label="Full data")
        ax_sktime = axs[degree, 1]
        ax_sktime.scatter(df.index, df[target_col], color="lightgray", s=5, alpha=0.6, label="Full data")
        
        # sklearn
        pipe = make_pipeline(
            PolynomialFeatures(degree=degree, include_bias=True),
            LinearRegression(fit_intercept=False),
        )
        pipe.fit(df["time_int"].values.reshape(-1, 1), df[target_col].values)
        y_trend_sklearn = pipe.predict(df["time_int"].values.reshape(-1, 1))

        ax_sklearn.plot(
            df.index,
            y_trend_sklearn,
            color="red",
            linestyle="-",
            linewidth=3,
            label="Sklearn trend",
            alpha=0.85,
        )
        
        # sktime
        trend_forecaster = PolynomialTrendForecaster(degree=degree)
        trend_forecaster.fit(y_series)
        
        fh = ForecastingHorizon(df.index)
        y_trend_sktime = trend_forecaster.predict(fh)
        
        ax_sktime.plot(
            df.index,
            y_trend_sktime,
            color="blue",
            linestyle="-",
            linewidth=3,
            label="Sktime trend",
            alpha=0.85,
        )
        
        # Set titles and labels
        ax_sklearn.set_title(f"Sklearn - Polynomial Degree {degree}")
        ax_sktime.set_title(f"Sktime - Polynomial Degree {degree}")
        
        ax_sklearn.set_ylabel(target_col)
        ax_sktime.set_ylabel(target_col)
        
        ax_sklearn.legend(loc="upper left", fontsize=10)
        ax_sktime.legend(loc="upper left", fontsize=10)
        
    axs[-1, 0].set_xlabel("Date")
    axs[-1, 1].set_xlabel("Date")
    
    plt.tight_layout()
    plt.show()


df = create_data()
plot_polynomial_trend_comparison(df, max_degree=4)
```
<img width="1990" height="2490" alt="Image" src="https://github.com/user-attachments/assets/e30521cf-31d5-45a4-8855-c8b432e127fb" />

**Expected behavior**

Each polynomial degree should produce a different fitted trend, consistent with sklearn’s PolynomialFeatures + LinearRegression pipeline.

**Additional context**

In my actual dataset, this behavior already occurs starting from degree 1, meaning I can obtain at most a linear trend forecast. This suggests the bug may also be influenced by the dataset.

**Versions**
<details>

System:
    python: 3.10.16 (main, Jun 25 2025, 17:40:50) [Clang 17.0.0 (clang-1700.0.13.5)]
    machine: macOS-15.6-arm64-arm-64bit

Python dependencies:
          pip: 25.1.1
       sktime: 0.38.4
      sklearn: 1.7.0
       skbase: 0.12.3
        numpy: 1.26.4
        scipy: 1.15.3
       pandas: 2.3.0
   matplotlib: 3.10.3
       joblib: 1.5.1
        numba: 0.61.2
  statsmodels: 0.14.4
     pmdarima: 2.0.4
statsforecast: 2.0.1
      tsfresh: None
      tslearn: None
        torch: None
   tensorflow: None
</details>

**Comment by fkiraly** [comment]
Reproduced on python 3.12, windows.

Here is my printout for `regressor_`, strangely this shows the correct internal degree, meaning the wrapped pipeline is equal to the manually constructed pipeline:

```
Pipeline(steps=[('polynomialfeatures', PolynomialFeatures(degree=0)),
                ('linearregression', LinearRegression(fit_intercept=False))])
Pipeline(steps=[('polynomialfeatures', PolynomialFeatures(degree=1)),
                ('linearregression', LinearRegression(fit_intercept=False))])
Pipeline(steps=[('polynomialfeatures', PolynomialFeatures()),
                ('linearregression', LinearRegression(fit_intercept=False))])
Pipeline(steps=[('polynomialfeatures', PolynomialFeatures(degree=3)),
                ('linearregression', LinearRegression(fit_intercept=False))])
Pipeline(steps=[('polynomialfeatures', PolynomialFeatures(degree=4)),
                ('linearregression', LinearRegression(fit_intercept=False))])
```

Strange.

**Comment by fkiraly** [comment]
ok, I think I start to understand the issue - it is of numerical nature. This relates to the absolute magnitude of the data, where magnitudes that are too far away from the number 1 can cause error propagation in higher powers. Apparently, `sklearn` does not seem to manage this too well.

For instance, if we insert lines `X_sklearn = X_sklearn / 10000` in `_fit` and `_predict`, below is the plot we get - now degrees up to 3 agree:

<img width="1990" height="2490" alt="Image" src="https://github.com/user-attachments/assets/4dcf4b5c-91d0-42fe-924d-f181fb514558" />

**Comment by fkiraly** [comment]
... based on this, I start to wonder whether using the pipeline is such a good idea. Probably a directly implemented version of polynomial regression would be better.

The problem is that an estimator can be chosen which is not linear, not sure how to deal with that in a numerically stable way. Perhaps ensuring the index values are within [0 and 1]?

**Comment by SueP12** [comment]
I think you’re right. My dataset was hourly data over multiple years, and when I modified create_data() to use hourly data, the issue also appeared from degree 1.
I changed _get_X_numpy_int_from_pandas(x) to:
```python
def _get_X_numpy_int_from_pandas(x):
    return np.arange(len(x)).reshape(-1, 1)
```
With this change, both plots now show the same behavior.

**Comment by fkiraly** [comment]
> I think you’re right. My dataset was hourly data over multiple years, and when I modified create_data() to use hourly data, the issue also appeared from degree 1. I changed _get_X_numpy_int_from_pandas(x) to:
> 
> def _get_X_numpy_int_from_pandas(x):
>     return np.arange(len(x)).reshape(-1, 1)
> 
> With this change, both plots now show the same behavior.

The problem with this change is that the coefficients will be in iloc (integer) units rather than loc (time), which would be a breaking change for people who use coefficients.

Also, for unequally spaced time stamps, you would now get a different model.

I think the difficult question is, can we resolve the issue with numerical stability (or instability) without impacting the coefficients?

**Comment by aryanyk** [comment]
Hi @fkiraly , I’d like to work on this issue.

From the discussion, it seems the behavior may be due to numerical instability when using large-magnitude time indices in polynomial features. I’m considering exploring an approach where the input is rescaled or normalized internally to improve stability, while ensuring that the existing behavior and coefficient interpretation are preserved.

Would this direction be acceptable, or is there a preferred approach you’d recommend?
